### ISSUE TITLE
Enable pre and post content to be added

### ISSUE THREAD

**Issue by simeg** [YouCanBookMe/react-datetime#153]
Reading the comments for #45 there's a suggestion to add prefix and postfix properties to enable to input content on either left (pre) or right (post) side of the input field. 

To me this seems like a reasonable approach to enable users to add e.g. icons, buttons and similar things (that's often requested). This way we're not changing any code of the actual component.

A downside to this is of course that the CSS could break due to changing the HTML, but that is a risk I think is OK.

I'm interested in hearing people's opinions about this (especially @arqex's).

**Comment by arqex** [comment]
Hey @simeg 

I had thought something even more generic. We can have a `renderInput` prop that accepts a function like:

```js
function renderInput( inputProps ){
 return <input { ...inputProps } />
}
```

That would print the input just right like it is now, so it would be our default `renderInput` prop. But that is so flexible, people could customize it so much that they could render whatever they like, not even an input there.

We could do the same with the views, `renderView` would receive the type of the view to render:

```js
function renderView( viewType, ViewComponent, viewProps ){
 return <ViewComponent { ...viewProps } />
}
```

Again the developer could completely customize every view (years, months, days, time), changing it slightly or building a completely new view.

**Comment by simeg** [comment]
This was implemented in #464, looks like we didn't have to wait for v3!
